packages/serialization/form/test/common/formParseNode.ts

```diff
@@ -0,0 +1,52 @@
+import { assert } from "chai";
+
+import { FormParseNode } from "../../src/index";
+import { createTestEntityFromDiscriminator, TestEntity } from "../testEntity";
+
+describe("FormParseNode", () => {
+  const testUserForm =
+    "displayName=Megan+Bowen&" +
+    "numbers=one,two,thirtytwo&" +
+    "givenName=Megan&" +
+    "accountEnabled=true&" +
+    "createdDateTime=2017-07-29T03:07:25Z&" +
+    "jobTitle=Auditor&" +
+    "mail=MeganB@M365x214355.onmicrosoft.com&" +
+    "mobilePhone=null&" +
+    "officeLocation=null&" +
+    "preferredLanguage=en-US&" +
+    "surname=Bowen&" +
+    "workDuration=PT1H&" +
+    "startWorkTime=08:00:00.0000000&" +
+    "endWorkTime=17:00:00.0000000&" +
+    "userPrincipalName=MeganB@M365x214355.onmicrosoft.com&" +
+    "birthday=2017-09-04&" +
+    "id=48d31887-5fad-4d73-a9f5-3c356e68a038";
+  it("getsEntityValueFromForm", () => {
+    const parseNode = new FormParseNode(testUserForm);
+    const testEntity = parseNode.getObjectValue(
+      createTestEntityFromDiscriminator
+    ) as TestEntity;
+    assert.isNotNull(testEntity);
+    assert.isUndefined(testEntity.officeLocation);
```

Comment:
Should this be undefined or null?
I ask as undefined can be the result of the property existing and being set as undefined or the property itself not existing.
The null case can only every happen if the property exists.
If undefined is the right value, do we need an addition assertion on `testEntity.additionalData.hasOwnProperty("mobilePhone")`?

Reply:
We're only returning undefined, never null, throughout the kiota core libs, to avoid confusion. I'm not against reviewing that assumption, but I think this should be done separately from this effort.

Reply:
So, are we returning undefined here because the property exists and is assigned the undefined value?

Reply:
Actually, I don't think it has any material difference. I was thinking about the case where you want to delete the property on the server by sending null, which is only relevant when you're mutating the deserialized object and serializing it back over the wire to the server.
I'm good here, thanks.

Reply:
yeah this is to parse null values (which we do in JSON and TEXT) and not return "null" (string value)